invenio_communities/communities/services/components.py

```diff
@@ -300,6 +300,20 @@ def update(self, identity, data=None, record=None, errors=None, **kwargs):
                 record["theme"] = data["theme"]
 
 
+class CommunityParentComponent(ServiceComponent):
+    """Service Component for Community parent."""
+
+    def update_parent(self, identity, data=None, record=None, **kwargs):
+        """Update parent community of a community."""
+        try:
+            parent = self.service.record_cls.pid.resolve(data)
+            if parent.parent:
+                raise ValidationError("Child community cannot be parent.")
+        except PIDDoesNotExistError:
+            raise ValidationError("Parent community does not exist.")
+        record.parent = parent
```

Comment:
minor: since we have a dedicated method on the service that "does the thing it say", it would be better to have the logic there (since components are about "side-effects" and isolating their business logic).

If we move to use the `update` service method, this code can stay. We should also handle other methods then, for sure `create`.